lib/cli/src/Cardano/CLI.hs

```diff
@@ -1303,19 +1308,17 @@ tlsOption = TlsConfiguration
         <> help "The RSA Server key which signed the x.509 server certificate."
 
 smashURLOption
-    :: Parser URI
+    :: Parser PoolMetadataSource
 smashURLOption = option (eitherReader reader) $ mempty
     <> long "smash-url"
     <> metavar "URL"
-    <> help "Optional HTTP(S) address of a \
-            \Stakepool Metadata Aggregation Server."
+    <> help "Source of SMASH (Stakepool Metadata Aggregation Server). Possible values: \
```

Comment:
That is not quite right. The option does not capture the source of SMASH, it captures the strategy for fetching stake pool metadata. As a matter of fact, `none` and `direct` makes no use of SMASH. 

`pool-metadata-fetching-strategy` would better reflect what that option is, although it's arguably a long name so if you have a better suggestion :grimacing: